lib/src/Pos/Logic/Pure.hs

```diff
@@ -0,0 +1,314 @@
+-- data GenericBlock b = UnsafeGenericBlock
+--     { _gbHeader :: !(GenericBlockHeader b)
+--     , _gbBody   :: !(Body b)
+--     , _gbExtra  :: !(ExtraBodyData b)
+--     } deriving (Generic)
+--
+-- data GenericBlockHeader b = UnsafeGenericBlockHeader
+--     { -- | Pointer to the header of the previous block.
+--       _gbhPrevBlock :: !(BHeaderHash b)
+--     , -- | Proof of body.
+--       _gbhBodyProof :: !(BodyProof b)
+--     , -- | Consensus data to verify consensus algorithm.
+--       _gbhConsensus :: !(ConsensusData b)
+--     , -- | Any extra data.
+--       _gbhExtra     :: !(ExtraHeaderData b)
+--     } deriving (Generic)
+
+
+-- | This block is always given by 'getBlock' and 'getTip'
+block :: Block
+block = Right mainBlock
+
+mainBlock :: MainBlock
+mainBlock = UnsafeGenericBlock
+    { _gbHeader = mainBlockHeader
+    , _gbBody   = blockBody
+    , _gbExtra  = extraBodyData
+    }
+
+blockBody :: Body MainBlockchain
+blockBody = MainBody
+    { _mbTxPayload     = emptyTxPayload
+    , _mbSscPayload    = emptySscPayload
+    , _mbDlgPayload    = emptyDlgPayload
+    , _mbUpdatePayload = emptyUpdatePayload
+    }
+
+emptyTxPayload :: TxPayload
+emptyTxPayload = UnsafeTxPayload
+    { _txpTxs       = MerkleEmpty
+    , _txpWitnesses = []
+    }
+
+-- SscPayload is 4 alternatives. I chose CertificatesPayload because it has
+-- the fewest fields...
+emptySscPayload :: SscPayload
+emptySscPayload = CertificatesPayload
+    { spVss = UnsafeVssCertificatesMap
+          { getVssCertificatesMap = mempty
+          }
+    }
+
+emptyDlgPayload :: DlgPayload
+emptyDlgPayload = UnsafeDlgPayload
+    { getDlgPayload = []
+    }
+
+emptyUpdatePayload :: UpdatePayload
+emptyUpdatePayload = UpdatePayload
+    { upProposal = Nothing
+    , upVotes    = []
+    }
+
+extraBodyData :: ExtraBodyData MainBlockchain
+extraBodyData = MainExtraBodyData
+    { _mebAttributes = Attributes
+          { attrData   = ()
+          , attrRemain = UnparsedFields mempty
+          }
+    }
+
+blockHeader :: BlockHeader
+blockHeader = Right mainBlockHeader
+
+mainBlockHeader :: MainBlockHeader
+mainBlockHeader = UnsafeGenericBlockHeader
+    { _gbhPrevBlock = mainBlockHeaderHash
+    , _gbhBodyProof = bodyProof
+    , _gbhConsensus = consensusData
+    , _gbhExtra     = extraHeaderData
+    }
+
+mainBlockHeaderHash :: HeaderHash
+mainBlockHeaderHash = AbstractHash (Crypto.hash (mempty :: ByteString))
+
+bodyProof :: BodyProof MainBlockchain
+bodyProof = MainProof
+    { mpTxProof       = txProof
+    , mpMpcProof      = sscProof
+    , mpProxySKsProof = dlgProof
+    , mpUpdateProof   = updateProof
+    }
+
+txProof :: TxProof
+txProof = TxProof
+    { txpNumber        = 0
+    , txpRoot          = MerkleRoot (AbstractHash (Crypto.hash (mempty :: ByteString)))
+    , txpWitnessesHash = AbstractHash (Crypto.hash (mempty :: ByteString))
+    }
+
+sscProof :: SscProof
+sscProof = CertificatesProof
+    { sprVss = AbstractHash (Crypto.hash (mempty :: ByteString))
```

Comment:
Please avoid `Crypto.hash`. The whole point of `Pos.Crypto.Hashing` is to encapsulate hashing library. `unsafeHash` works fine in this case. Or `hashRaw`.